### DISCUSSION TITLE
Adding new build targets to meson

### DISCUSSION THREAD

**Discussion by mahmnv** [mesonbuild/meson#12236] [Q&A]
Hi,

I'm working on adding support for a custom compiler to meson. build targets like `executable` does not represent output files of this compiler and I'd like to add some custom build targets like `jar` exists for java. Commit which adds `jar` is quite old and lot of incremental refactoring has gone into meson after that. Currently, what are the files that need to be modified for adding such a build target?

**Comment by eli-schwartz** [comment]
- `mesonbuild/interpreter/interpreter.py` to add a new function to meson.build
- `mesonbuild/build.py` to add a new backend abstraction such as the `Target` / `Executable` / `Jar`
- `mesonbuild/backend/backends.py` to hook up rules generation
- `mesonbuild/compilers/` to add a new compiler language type, see detect.py for handling add_language and create a new file named like the language containing the compiler class itself.


What's the compiler in question by the way? :)

**Reply by mahmnv** [reply]
Thanks for the answer. I'll keep this open while I work on it. Sorry I cant tell you the compiler name right now :)
